frontend/src/metabase/query_builder/components/MoveQuestionModal/MoveQuestionModal.tsx

```diff
@@ -0,0 +1,223 @@
+
+import { useUpdateCardMutation } from "metabase/api";
+import { QuestionMoveConfirmModal } from "metabase/collections/components/CollectionBulkActions/QuestionMoveConfirmModal";
+import type { MoveDestination } from "metabase/collections/types";
+import { canonicalCollectionId } from "metabase/collections/utils";
+import ConfirmContent from "metabase/components/ConfirmContent";
+import Modal from "metabase/components/Modal";
+import { MoveModal } from "metabase/containers/MoveModal";
+import Dashboards from "metabase/entities/dashboards";
+import { INJECT_RTK_QUERY_QUESTION_VALUE } from "metabase/entities/questions";
+import { useDispatch } from "metabase/lib/redux";
+import * as Urls from "metabase/lib/urls";
+import { API_UPDATE_QUESTION } from "metabase/query_builder/actions";
+import QuestionMoveToast from "metabase/questions/components/QuestionMoveToast";
+import { addUndo } from "metabase/redux/undo";
+import { Box, Icon, Radio, Title } from "metabase/ui";
+import type Question from "metabase-lib/v1/Question";
+
+interface MoveQuestionModalProps {
+  question: Question;
+  onClose: () => void;
+}
+
+type ConfirmationTypes =
+  | "dashboard-to-dashboard"
+  | "dashboard-to-collection"
+  | "collection-to-dashboard";
+
+export const MoveQuestionModal = ({
+  question,
+  onClose,
+}: MoveQuestionModalProps) => {
+  const dispatch = useDispatch();
+
+  const [updateQuestion] = useUpdateCardMutation();
+
+  const [confirmMoveState, setConfirmMoveState] = useState<{
+    type: ConfirmationTypes;
+    destination: MoveDestination;
+    affectedDashboards?: any[];
+    isLoading?: boolean;
+  } | null>(null);
+  const [deleteOldDashcards, setDeleteOldDashcards] = useState<
+    boolean | undefined
+  >();
+
+  const handleMove = async (destination: MoveDestination) => {
+    const update =
+      destination.model === "dashboard"
+        ? { dashboard_id: destination.id as number }
+        : {
+            dashboard_id: null,
+            collection_id: canonicalCollectionId(destination.id),
+          };
+
+    await updateQuestion({
+      id: question.id(),
+      delete_old_dashcards: deleteOldDashcards,
+      ...update,
+    })
+      .then(({ data: updatedCard }) => {
+        // HACK: entity framework would previously keep the qb in sync
+        // with changing where the question lived
+        dispatch({ type: API_UPDATE_QUESTION, payload: updatedCard });
+        dispatch({
+          type: INJECT_RTK_QUERY_QUESTION_VALUE,
+          payload: updatedCard,
+        });
+
+        dispatch(
+          addUndo({
+            message: (
+              <QuestionMoveToast
+                destination={destination}
+                question={question}
+              />
+            ),
+            undo: false,
+          }),
+        );
+
+        if (destination.model === "dashboard") {
+          dispatch(
+            push(
+              Urls.dashboard(
+                { id: destination.id, name: "TODO" },
```

Comment:
An empty string is likely fine here. I'll fix it.

Reply:
Not if I fix it first